lib/containers/engine.pm

```diff
@@ -0,0 +1,396 @@
+# SUSE's openQA tests
+#
+# Copyright © 2020-2021 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Abstraction layer to operate docker and podman containers through same interfaces
+# Maintainer: qac team <qa-c@suse.de>
+
+package containers::engine;
+use Mojo::Base -base;
+use testapi;
+use Test::Assert 'assert_equals';
+use containers::utils qw(registry_url);
+use utils qw(systemctl file_content_replace);
+
+has runtime => undef;
+
+sub _engine_assert_script_run {
+    my ($self, $cmd, @args) = @_;
+    assert_script_run($self->runtime . " " . $cmd, @args);
+}
+
+sub _engine_script_run {
+    my ($self, $cmd, @args) = @_;
+    return script_run($self->runtime . " " . $cmd, @args);
+}
+
+sub _engine_script_output {
+    my ($self, $cmd, @args) = @_;
+    return script_output($self->runtime . " " . $cmd, @args);
+}
+
+=head2 create_container($image, $name, [$cmd])
+
+Creates a container.
+C<image> the name of the image we create the container from.
+C<name> is the given name of the created container.
+C<cmd> can be anything you want to run in the container, similar to C<run>. for 
+instance: C<docker create -it tumbleweed bash>
+
+=cut
+sub create_container {
+    my ($self) = shift;
+    my %args = (
+        image => '',
```

Comment:
do you really need these lines? I think you can just have
`my ($self, %args);`
and then the `die` checks you have below. No need for these things here IMO.

Reply:
this is a better practice when you wanna use named variables. and makes the code more readable